packages/wrangler/src/pipelines/index.ts

```diff
@@ -0,0 +1,459 @@
+						type: "string",
+						demandOption: true,
+					})
+					.option("r2", {
+						type: "string",
+						describe: "Destination R2 bucket name",
+						demandOption: true,
+					});
+			},
+			async (args) => {
+				await printWranglerBanner();
+
+				const config = readConfig(args.config, args);
+				const bucket = args.r2;
+				const name = args.pipeline;
+				const compression =
+					args.compression === undefined ? "gzip" : args.compression;
+
+				const batch = {
+					max_mb: args["batch-max-mb"],
+					max_duration_s: args["batch-max-seconds"],
+					max_rows: args["batch-max-rows"],
+				};
+
+				if (!bucket) {
+					throw new FatalError("Requires a r2 bucket");
+				}
+
+				const accountId = await requireAuth(config);
+
+				const pipelineConfig: PipelineUserConfig = {
+					name: name,
+					metadata: {},
+					source: [
+						{
+							type: "http",
+							format: "json",
+						},
+						{
+							type: "binding",
+							format: "json",
+						},
+					],
+					transforms: [],
+					destination: {
+						type: "r2",
+						format: "json",
+						compression: {
+							type: compression,
+						},
+						batch: batch,
+						path: {
+							bucket: bucket,
+						},
+						credentials: {
+							endpoint: getAccountR2Endpoint(accountId),
+							access_key_id: args["access-key-id"] || "",
+							secret_access_key: args["secret-access-key"] || "",
+						},
+					},
+				};
+				const destination = pipelineConfig.destination;
+				if (
+					!destination.credentials.access_key_id &&
+					!destination.credentials.secret_access_key
+				) {
+					// auto-generate a service token
+					await authorizeR2Bucket(name, accountId, pipelineConfig.destination);
+				}
+
+				if (!destination.credentials.access_key_id) {
+					throw new FatalError("Requires a r2 access key id");
+				}
+
+				if (!destination.credentials.secret_access_key) {
+					throw new FatalError("Requires a r2 secret access key");
+				}
+
+				if (args.authentication !== undefined && args.authentication === true) {
+					pipelineConfig.source = [
+						{
+							type: "binding",
+							format: "json",
+						},
+					];
+				}
+
+				if (args.transform !== undefined) {
+					const split = args.transform.split(".");
+					if (split.length === 2) {
+						pipelineConfig.transforms.push({
+							script: split[0],
+							entrypoint: split[1],
+						});
+					} else if (split.length === 1) {
+						pipelineConfig.transforms.push({
+							script: split[0],
+							entrypoint: "Transform",
+						});
+					} else {
+						throw new Error(
+							"invalid transform: required syntax script.entrypoint"
+						);
+					}
+				}
+
+				if (args.filepath) {
+					pipelineConfig.destination.path.filepath = args.filepath;
+				}
+				if (args.filename) {
+					pipelineConfig.destination.path.filename = args.filename;
+				}
+
+				logger.log(`🌀 Creating pipeline named "${name}"`);
+				const pipeline = await createPipeline(accountId, pipelineConfig);
```

Comment:
What should happen if this fails?

Reply:
It throws an exception with the appropriate error message.  Is there a better mechanism for errors?  This seemed to be the common pattern among other handlers.